app/containers/HomePage/sagas.js

```diff
@@ -2,43 +2,54 @@
  * Gets the repositories of the user from Github
  */
 
-import { take, call, put, select, race } from 'redux-saga/effects';
-
+import { take, call, put, select, fork, cancel } from 'redux-saga/effects';
 import { LOCATION_CHANGE } from 'react-router-redux';
-
 import { LOAD_REPOS } from 'containers/App/constants';
 import { reposLoaded, repoLoadingError } from 'containers/App/actions';
 
 import request from 'utils/request';
 import { selectUsername } from 'containers/HomePage/selectors';
 
-// Individual exports for testing
-export function* getGithubData() {
-  while (true) { // eslint-disable-line no-constant-condition
-    const watcher = yield race({
-      loadRepos: take(LOAD_REPOS),
-      stop: take(LOCATION_CHANGE), // stop watching if user leaves page
-    });
-
-    if (watcher.stop) break;
-
-    const username = yield select(selectUsername());
-    const requestURL = `https://api.github.com/users/${username}/repos?type=all&sort=updated`;
-
-    // Use call from redux-saga for easier testing
-    const repos = yield call(request, requestURL);
-
-    // We return an object in a specific format, see utils/request.js for more information
-    if (repos.err === undefined || repos.err === null) {
-      yield put(reposLoaded(repos.data, username));
-    } else {
-      console.log(repos.err.response); // eslint-disable-line no-console
-      yield put(repoLoadingError(repos.err));
-    }
+/**
+ * Github repos request/response handler
+ */
+export function* getRepos() {
+  // Select username from store
+  const username = yield select(selectUsername());
+  const requestURL = `https://api.github.com/users/${username}/repos?type=all&sort=updated`;
+
+  // Call our request helper (see 'utils/request')
+  const repos = yield call(request, requestURL);
+
+  if (!repos.err) {
+    yield put(reposLoaded(repos.data, username));
+  } else {
+    yield put(repoLoadingError(repos.err));
+  }
+}
+
+/**
+ * Watches for LOAD_REPOS action and calls handler
+ */
+export function* getReposWatcher() {
+  while (yield take(LOAD_REPOS)) {
+    yield call(getRepos);
   }
 }
 
+/**
+ * Root saga manages watcher lifecycle
+ */
```

Comment:
it may be worth noting that this is the point of contact between saga and route handler, since this seems to be the cause of a lot of confustion